src/driver/src/DefaultDriver.cpp

```diff
@@ -241,9 +100,9 @@ DefaultDriver::OutcomeCode doRunLogic(const DefaultDriver::ProgramOptions& optio
         phaseConfigSource = fs::path(options.workloadSource).parent_path();
     }
 
-    YamlParameters params;
-    auto config = loadConfig(options.workloadSource, options.workloadSourceType);
-    auto yaml = recursiveParse(config, params, phaseConfigSource);
+    WorkloadParser parser{phaseConfigSource, options.isSmokeTest};
+
+    auto yaml = parser.parse(options.workloadSource, options.workloadSourceType);
```

Comment:
instead of peeling off 2 things from `options` does `parser.parse()` want to just take in the whole `options`?

Reply:
👍 please put this comment in there or somesuch - `// consider passing in whole options struct if we pass in more than 2-3 fields`